waiter/integration/waiter/token_request_test.clj

```diff
@@ -1085,32 +1085,46 @@
 (deftest ^:parallel ^:integration-fast test-token-https-redirects
   (testing-using-waiter-url
     (let [token (rand-name)
-          token-response (post-token waiter-url {:cmd (str (kitchen-cmd) " -p $PORT0")
-                                                 :cmd-type "shell"
-                                                 :https-redirect true
-                                                 :name token
-                                                 :version "does-not-matter"
-                                                 :token token})]
-      (assert-response-status token-response http-200-ok)
-
-      (testing "https redirects"
-        (let [request-headers {:x-waiter-token token}
-              url (URL. (str "http://" waiter-url))
-              endpoint "/request-info"]
-
-          (testing "get request"
-            (let [{:keys [headers] :as response}
-                  (make-kitchen-request waiter-url request-headers :method :get :path endpoint)]
-              (assert-response-status response http-301-moved-permanently)
-              (is (= (str "https://" (.getHost url) endpoint) (get headers "location")))
-              (is (str/starts-with? (str (get headers "server")) "waiter") (str "headers:" headers))))
-
-          (testing "post request"
-            (let [{:keys [headers] :as response}
-                  (make-kitchen-request waiter-url request-headers :method :post :path endpoint)]
-              (assert-response-status response http-307-temporary-redirect)
-              (is (= (str "https://" (.getHost url) endpoint) (get headers "location")))
-              (is (str/starts-with? (str (get headers "server")) "waiter") (str "headers:" headers))))))
+          token-description (assoc (kitchen-params) :https-redirect true :name token :token token)]
+
+      (let [request-headers {"accept" "application/json"
+                             :x-waiter-token token}
+            url (URL. (str "http://" waiter-url))
+            endpoint "/request-info"
+            ping-endpoint "/waiter-ping"]
+        (doseq [request-method [:get :post]]
+          (let [redirect-status-code (if (= :get request-method) http-301-moved-permanently http-307-temporary-redirect)]
```

Comment:
I think it's more clear to just iterate over pairs rather than having the `if` inside to determine the status code.

```suggestion
        (doseq [[request-method redirect-status-code] [[:get http-301-moved-permanently] [:post http-307-temporary-redirect]]]
```

Reply:
Done.